name: endtest
on:
  workflow_dispatch:
  pull_request:
    branches: [search] #You would need to make sure this is correct every time

jobs:
  run_test_case:
    name: Run test case and check for errors
    runs-on: ubuntu-latest
    #needs: Ideally a job above this will prevent this from running. TBD
    steps:
      - uses: actions/checkout@master
      - name: Running EndTest test case(s)
        id: endtest_functional_tests
        uses: endtest-technologies/github-run-tests-action@v1.8
#        env: "not required"
#          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          app_id: 23844484
          app_code: 85029485
          api_request: https://app.endtest.io/api.php?action=runWeb&appId=23844484&appCode=85029485&suite=869550&platform=windows&os=windows10&browser=chrome&browserVersion=latest&resolution=1920x1200&geolocation=sanfrancisco&cases=8329999,8329998&notes=
          number_of_loops: 10
      - name: Use the outputs from test execution in a different step #this section is not needed. These are just the variables to use
        run: |
          echo ${{ steps.endtest_functional_tests.outputs.test_suite_name }}
          echo ${{ steps.endtest_functional_tests.outputs.configuration }}
          echo ${{ steps.endtest_functional_tests.outputs.test_cases }}
          echo ${{ steps.endtest_functional_tests.outputs.passed }}
          echo ${{ steps.endtest_functional_tests.outputs.failed }}
          echo ${{ steps.endtest_functional_tests.outputs.errors }}
          echo ${{ steps.endtest_functional_tests.outputs.start_time }}
          echo ${{ steps.endtest_functional_tests.outputs.end_time }}
          echo ${{ steps.endtest_functional_tests.outputs.detailed_logs }}
          echo ${{ steps.endtest_functional_tests.outputs.screenshots_and_video }}
          echo ${{ steps.endtest_functional_tests.outputs.hash }}
          echo ${{ steps.endtest_functional_tests.outputs.results }}
      - name: Check for errors
        if: steps.endtest_functional_tests.outputs.errors != 0
        run: |
          echo "Please review, ${{ steps.endtest_functional_tests.outputs.failed }} error(s) occurred. View results here: ${{ steps.endtest_functional_tests.outputs.results }}"; exit 1;
          
      - name: Check for failed test cases
        if: steps.endtest_functional_tests.outputs.failed != 0
        run: |
          echo "Please review, ${{ steps.endtest_functional_tests.outputs.failed }} test case(s) failed. View results here: ${{ steps.endtest_functional_tests.outputs.results }}"; exit 1;